### PR TITLE
Support insert string data

### PR DESCRIPTION
**Pull request by xige-16**
issue: #15888 
/kind: feature
Signed-off-by: xige-16 <xi.ge@zilliz.com>

### CODE REVIEW

**Comment by jennyli-z** [comment]
rerun ut

File: internal/storage/stats.go [hunk 1]
```diff
@@ -29,12 +33,78 @@ const (
...
+	case *schemapb.ID_LongPk:
+		if pkID.GetLongPk() < minID.GetLongPk() {
+			stats.MinPk = pk
+		}
+		if maxID.GetLongPk() < pkID.GetLongPk() {
+			stats.MaxPk = pk
+		}
+
+	case *schemapb.ID_StringPk:
+		if pkID.GetStringPk() < minID.GetStringPk() {
```
Comment (xiaofan-luan) [inline internal/storage/stats.go:99]: the string PK compare function might be wrong for ""

File: internal/querynode/flow_graph_filter_dm_node.go [hunk 2]
```diff
@@ -145,7 +145,7 @@ func (fdmNode *filterDmNode) filterInvalidDeleteMessage(msg *msgstream.DeleteMsg
 
 // filterInvalidInsertMessage would filter out invalid insert messages
 func (fdmNode *filterDmNode) filterInvalidInsertMessage(msg *msgstream.InsertMsg) *msgstream.InsertMsg {
-	if !msg.CheckAligned() {
+	if err := msg.CheckAligned(); err != nil {
 		// TODO: what if the messages are misaligned? Here, we ignore those messages and print error
 		log.Warn("Error, misaligned messages detected")
```
[the remark this answers is not in the document]
Comment (xige-16) [reply internal/querynode/flow_graph_filter_dm_node.go:150]: > What happens if CheckAligned failed? do we simply throw the message?

This is the error that appears in the flowgraph, and sdk can't get the error druing inserting data.

[the remark this answers is not in the document]
Comment (xige-16) [reply internal/querynode/flow_graph_filter_dm_node.go:150]: The proxy has already checkAligned when sending insert msg, this error should not appear

File: internal/datanode/segment_replica.go [hunk 3]
```diff
@@ -107,23 +109,126 @@ type SegmentReplica struct {
...
+			s.maxPK = id
 		}
-		if pk < s.minPK {
-			s.minPK = pk
+	case *schemapb.ID_StringPk:
+		oldMinPk := s.minPK.GetStringPk()
+		oldMaxPk := s.minPK.GetStringPk()
+		newPk := id.GetStringPk()
+		//TODO::xige-16, opt string compare
+		if len(oldMinPk) == 0 || newPk < oldMinPk {
```
[the remark this answers is not in the document]
Comment (xige-16) [reply internal/datanode/segment_replica.go]: has modified with the first method

File: internal/storage/stats.go (hunk 1 above)
Comment (xige-16) [reply internal/storage/stats.go]: use strings pkg to compare(strings.Compare(str1, str2)

**Pull request merged**